utils/src/main/java/org/corfudb/utils/lock/LockClient.java

```diff
@@ -154,6 +183,30 @@ private void monitorLocks() {
         ));
     }
 
+
+    /**
+     * Stop monitoring the locks.
+     */
+    private void stopMonitorLocks() {
+        synchronized (lockMonitorFuture) {
+            if (lockMonitorFuture.isPresent() && !lockMonitorFuture.get().isDone()) {
+                boolean cancelled = lockMonitorFuture.get().cancel(false);
+                if (!cancelled) {
+                    log.error("Lock: Could not cancel the future for lease renewal!!!", lock.getLockId());
+                }
+            }
+            log.warn("Lock: canceled the future for monitor locks!!!", lock.getLockId());
+        }
+    }
+
+
+    public void stop() {
+        lock.input(LockEvent.UNRECOVERABLE_ERROR);
```

Comment:
The stop() could have been called in a non-error scenario also.  Can we add another event instead of unrecoverable error?